File: 04_quicksort/swift/02_recursive_sum.swift
// we read standard input line by line
func sum(arr: [Int]) -> Int {
	if arr.count == 0{
		return 0
	}
	let temp = arr[0]
	var newArr = arr
	newArr.removeFirst()
	return temp + sum(arr: newArr)
}

print(sum(arr: [1, 2, 3, 4]))

File: 04_quicksort/c++/02_recursive_sum.cpp
#include <iostream>
#include <vector>
using namespace std;

int sum(vector<int> arr){
	if (!arr.size())
		return 0;
	else{
		int temp = arr[0];
		arr.erase(arr.begin(), arr.begin()+1);
		return temp + sum(arr);
	}
}
int main(int argc, char *argv[]) {
	vector<int> arr{1, 2, 3, 4};
	cout << sum(arr) << endl;
}